app/events_blueprint/views.py

```diff
@@ -2,50 +2,54 @@
 from flask import Flask, jsonify,abort,request,session
 from flask import make_response
 from app import createApp
-from app.common_functions import token_required
+from app.common_scripts.common_functions import token_required
 from app.auth_blueprint import models as users_models
 
 from . import models
 from . import events
 
+"""Helper function to check for blacklisted tokens"""
+
+def check_blacklisted_token(token):
+
+    token = users_models.TokenBlackList.query.filter_by(token=token).first()
+    if token:
+        return True
+    return False
 
 """ HANDLE EVENTS ACTIVITIES """
 #create a new event
 @events.route('/api/v1/events', methods=['POST'])
 @token_required
-def add_event(logged_in_user):
-    #check if user is logged in
+def add_event(logged_in_user):    
+     #check if token is blacklisted
+    token = request.headers['x-access-token']
```

Comment:
if you have a decorator that handles the requirements of the token, why check for it again here?

Reply:
The decorator checks for expired tokens and invalid ones. Doesn't check for blacklisted tokens. Maybe I incorporate that in the decorator and remove the blacklist check from all endpoints used. Thanks, I hadn't thought of that.